### DISCUSSION TITLE
New feature request: add Makefile

### DISCUSSION THREAD

**Discussion by delgh1** [geerlingguy/ansible-for-devops-manuscript#5] [Ideas]
Hi Jeff Geerling, thanks for the amazing book. If I may, I have a new idea: if you add a Makefile that can generate a pdf and/or epub and/or htmls, that would be super great. Then people can just `git pull` and `make [...]` to get the newest book.

**Comment by geerlingguy** [comment]
That would be cool; I don't know if Leanpub would ever release their tooling for converting the markdown to PDF or other formats, but I know some people use tools like `markdown-pdf` to do it: https://superuser.com/questions/689056/how-can-i-convert-github-flavored-markdown-to-a-pdf

You would need to do some preprocessing to bundle everything together.

**Reply by buckeye43210** [reply]
I can give it a try, and while I'm at it, throw in the ability to render Mermaid diagrams :-)

**Reply by buckeye43210** [reply]
Initial stab at Makefile added to PR. Confirmed PDF and EPUB rendering.

Future enhancements include:
- LaTeX template for PDF output
- Book cover illustration
- Mermaid embedded diagram support

**Reply by buckeye43210** [reply]
Unfortunately, Pandoc does not currently support Markura extensions for syntax highlighting, width settings, and similar features :-1:

**Reply by buckeye43210** [reply]
Still, it does a decent job of rendering the manuscript into EPUB and PDF :+1:

**Reply by buckeye43210** [reply]
I may try setting up a sub directory for the Markua files and create a GitHub action leveraging Pandoc to convert GitHub Flavor Markdown (GFM) to Markua. The GFM displays better in GitHub and will work better when I add Mermaid (diagrams as code) to the chapters.
